arbor/include/arbor/simulation.hpp

```diff
@@ -34,6 +34,8 @@ class ARB_ARBOR_API simulation {
                const context& ctx=make_context(),
```

Comment:
should we update `simulation`'s ctors to take `context` by value since it is a `shared_ptr` now?

Reply:
Good catch!

Reply:
I see you pass `context` by value now everywhere. I like it.

Reply:
Well, it's the proper thing to do.